### PR TITLE
Fix for RT #131201 (error message when proto has no implementation)

### PR DESCRIPTION
**Pull request by jsimonet**
I tried to write some code to fix RT #131201.

Basically, it's a check to see if the array is empty or not, but i don't know how to inform user that a prototype method exists and may be implemented.

### CODE REVIEW

File: src/core/Exception.pm [hunk 1]
```diff
@@ -2462,9 +2462,10 @@ my class X::Multi::NoMatch is Exception {
                         ~  join("\n    ", '', sort keys @cand ∖ @un-rw-cand)
                     unless @cand == @un-rw-cand
                 )
-            !! join "\n    ",
-                'none of these signatures match:',
-                @cand
+            !! ( @cand
+                 ??  join "\n    ", 'none of these signatures match:', @cand
+                 !! "No signature match."
```
Comment (zoffixznet) [inline src/core/Exception.pm]: The `No signature match` isn't very clear. Perhaps, it should read `Routine does not have any candidates. Is only the proto defined?`

Comment (jsimonet) [inline src/core/Exception.pm]: Yes, it's a better message :) I'll commit this one.

**Comment by zoffixznet** [comment]
Thanks!

**Comment by zoffixznet** [comment]
For record, you fixed https://rt.perl.org/Ticket/Display.html?id=131367#ticket-history too

**Comment by jsimonet** [comment]
Perfect :) I'm happy to have contributed to Perl6!

**Pull request merged**